wire-schema/src/commonMain/kotlin/com/squareup/wire/schema/SchemaContext.kt

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2022 Block Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.squareup.wire.schema
+
+import com.squareup.wire.WireLogger
+import okio.Path
+
+interface SchemaContext {
+  /** Location on [fileSystem] where the [SchemaHandler] is to write files, if it needs to. */
+  val outDirectory: Path
```

Comment:
Shouldn't this be configured on the concrete `FileSystemContext` and not on the interface? How would this get used by a protoc plugin?

Reply:
I moved this and the directory creation under that schema context implementation